CMakeLists.txt

```diff
@@ -657,8 +657,11 @@ if (APPLE)
       target_link_libraries(shadps4 PRIVATE ${MOLTENVK})
   endif()
 
-  # Reserve system-managed memory space.
-  target_link_options(shadps4 PRIVATE -Wl,-no_pie,-no_fixup_chains,-no_huge,-pagezero_size,0x4000,-segaddr,TCB_SPACE,0x4000,-segaddr,GUEST_SYSTEM,0x400000,-image_base,0x20000000000)
+  # Using x86_64 (ARM by default if not set)
+  if (CMAKE_OSX_ARCHITECTURES)
```

Comment:
You will need to check more than just `CMAKE_OSX_ARCHITECTURES` because it may be compiled natively on an Intel Mac. Simplest option I think would be to move the logic added to `externals/CMakeList.txt` for ffmpeg to this file and use the `ARCHITECTURE` variable to see if `ARCHITECTURE STREQUAL "x86_64"`